samza-sql/src/main/java/org/apache/samza/sql/translator/ProjectTranslator.java

```diff
@@ -83,9 +104,20 @@ public SamzaSqlRelMessage apply(SamzaSqlRelMessage message) {
       for (int index = 0; index < output.length; index++) {
         names.add(index, project.getNamedProjects().get(index).getValue());
       }
-
+      updateMetrics(arrivalTime, Instant.now());
       return new SamzaSqlRelMessage(names, Arrays.asList(output));
     }
+
+    private void updateMetrics(Instant arrivalTime, Instant outputTime) {
+      messagesCounter.inc();
+      processingTimeHistogram.update(Duration.between(arrivalTime, outputTime).toMillis());
```

Comment:
Would a metric smaller than millis be better considering that for many jobs projection would be trivial processing ?

Reply:
discussed offline: unit need to be consistent across all operators, and milliseconds makes more sense for joins for example, so will keep it milliseconds (and might revisit this when we see actual  stats)